src/components/manage/Controlpanels/AddonsControlpanel.jsx

```diff
@@ -0,0 +1,384 @@
+  updatesAvailable: {
+    id: 'Updates available',
+    defaultMessage: 'Updates available',
+  },
+  updateInstalledAddons: {
+    id: 'Update installed addons:',
+    defaultMessage: 'Update installed addons:',
+  },
+  uninstall: {
+    id: 'Uninstall',
+    defaultMessage: 'Uninstall',
+  },
+});
+
+@injectIntl
+@connect(
+  (state, props) => ({
+    installedAddons: state.addons.installedAddons,
+    availableAddons: state.addons.availableAddons,
+    upgradableAddons: state.addons.upgradableAddons,
+    activeIndex: state.activeIndex,
+    pathname: props.location.pathname,
+  }),
+  dispatch => bindActionCreators({ installAddon, listAddons, uninstallAddon }, dispatch),
+)
+/**
+ * AddonsControlpanel class.
+ * @class AddonsControlpanel
+ * @extends Component
+ */
+export default class AddonsControlpanel extends Component {
+  /**
+   * Property types.
+   * @property {Object} propTypes Property types.
+   * @static
+   */
+  static propTypes = {
+    listAddons: PropTypes.func.isRequired,
+    installAddon: PropTypes.func.isRequired,
+    uninstallAddon: PropTypes.func.isRequired,
+    installedAddons: PropTypes.arrayOf(
+      PropTypes.shape({
+        '@id': PropTypes.string,
+        'id': PropTypes.string,
+        'title': PropTypes.string,
+        'version': PropTypes.string,
+        'description': PropTypes.string,
+        'upgrade_info': PropTypes.shape({
+          'available': PropTypes.boolean
+        })
+      })
+    ).isRequired,
+    availableAddons: PropTypes.arrayOf(
+      PropTypes.shape({
+        '@id': PropTypes.string,
+        'id': PropTypes.string,
+        'title': PropTypes.string,
+        'version': PropTypes.string,
+        'description': PropTypes.string,
+        'upgrade_info': PropTypes.shape({
+          'available': PropTypes.boolean
+        })
+      })
+    ).isRequired,
+    upgradableAddons: PropTypes.arrayOf(
+      PropTypes.shape({
+        '@id': PropTypes.string,
+        'id': PropTypes.string,
+        'title': PropTypes.string,
+        'version': PropTypes.string,
+        'description': PropTypes.string,
+        'upgrade_info': PropTypes.shape({
+          'available': PropTypes.boolean
+        })
+      })
+    ).isRequired,
+  };
+
+  /**
+   * Constructor
+   * @method constructor
+   * @param {Object} props Component properties
+   * @constructs Sharing
+   */
+  constructor(props) {
+    super(props);
+    this.onAccordionClick = this.onAccordionClick.bind(this);
+    this.onInstall = this.onInstall.bind(this);
+    this.onUninstall = this.onUninstall.bind(this);
+    this.state = {
+      activeIndex: -1,
+      installedAddons: [],
+      availableAddons: [],
+      upgradableAddons: [],
+    };
+  }
+
+  /**
+   * On accordion click handler
+   * @method onAccordionClick
+   * @param {object} event Event object.
+   * @returns {undefined}
+   */
+  onAccordionClick(event, titleProps) {
+    const { index } = titleProps
+    const { activeIndex } = this.state.activeIndex
```

Comment:
You don't need the `{` and `}` here since you don't want to destructure the variable. What this line now says is map the local `activeIndex` to `this.state.activeIndex.activeIndex`. You can use:
```
const { activeIndex } = this.state;
```